nilearn/_utils/niimg_conversions.py

```diff
@@ -33,6 +33,23 @@ def _check_same_fov(img1, img2):
             and np.allclose(img1.get_affine(), img2.get_affine()))
 
 
+def _assert_same_fov(**kwargs):
+    """ Assert the equivalence of all provided images. Parameter names are
+        used to generate user friendly error message.
+    """
+    errors = []
+    for (a_name, a_img), (b_name, b_img) in itertools.combinations(
```

Comment:
The use of itertools.combinations looks quite smart but you could just take the first image and make sure it has the same field of view as the remaining ones, couldn't you ?